test/TestUtilities/Test.Utility/TestServer/TcpListenerServer.cs

```diff
@@ -53,6 +70,52 @@ public async Task<T> ExecuteAsync<T>(Func<string, Task<T>> action)
                 CancellationToken.None);
         }
 
+        public TcpListenerServer()
+        {
+            _tlsCertificate = GenerateSelfSignedCertificate();
+        }
+
+        private static X509Certificate2 GenerateSelfSignedCertificate()
+        {
+            using (var rsa = RSA.Create(2048))
+            {
+                var request = new CertificateRequest("cn=test", rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+                var start = DateTime.UtcNow;
+                var end = DateTime.UtcNow.AddYears(1);
+                var cert = request.CreateSelfSigned(start, end);
+                var certBytes = cert.Export(X509ContentType.Pfx, "password");
+
+                return new X509Certificate2(certBytes, "password", X509KeyStorageFlags.Exportable);
+            }
+        }
+
+        private async Task StartInvalidTlsCertificateServer(TcpListener tcpListener, CancellationToken token)
+        {
+            while (!token.IsCancellationRequested)
+            {
+                using (var client = await Task.Run(tcpListener.AcceptTcpClientAsync, token))
+                using (var sslStream = new SslStream(client.GetStream(), false))
+                {
+                    sslStream.AuthenticateAsServer(_tlsCertificate, clientCertificateRequired: false, SslProtocols.Tls12, checkCertificateRevocation: true);
+                    using (var reader = new StreamReader(sslStream, Encoding.ASCII, false, 1))
+                    using (var writer = new StreamWriter(sslStream, Encoding.ASCII, 1, false))
+                    {
+                        while (!string.IsNullOrEmpty(reader.ReadLine()))
```

Comment:
Instead of looping, would reading the entire stream at once work?

```suggestion
                        reader.ReadToEnd();
```
https://learn.microsoft.com/en-us/dotnet/api/system.io.streamreader.readtoend?view=netframework-4.7.2

Reply:
No longer applicable based on other comments.